File: ruby/core/lib/mullet/container.rb
module Mullet

  # Collection of renderers which will be rendered in the order they were added.
  module Container
    def initialize()
      super if defined?(super)
      @children = []
    end

    def add_child(child)
      @children.push(child)
    end

    def delete_child(child)
      @children.delete(child);
    end

    def children()
      return @children
    end

    def clear_children()
      @children.clear()
    end

    # Renders children in order they were added.
    # 
    # @param [RenderContext] render_context
    #           render context
    def render_children(render_context)
      @children.each do |child|
        child.render(render_context)
      end
    end
  end

end

File: java/core/src/main/java/com/github/pukkaone/mullet/Container.java
/*
Copyright (c) 2011, Chin Huang
All rights reserved.

Redistribution and use in source and binary forms, with or without
modification, are permitted provided that the following conditions are met:

  * Redistributions of source code must retain the above copyright notice,
    this list of conditions and the following disclaimer.
  * Redistributions in binary form must reproduce the above copyright notice,
    this list of conditions and the following disclaimer in the documentation
    and/or other materials provided with the distribution.

THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
(INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON
ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
(INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
*/
package com.github.pukkaone.mullet;

import java.util.ArrayList;
import java.util.List;

/**
 * Collection of renderers which will be rendered in sequence.
 */
public abstract class Container {

    private List<Renderer> children = new ArrayList<Renderer>();

    protected Container() {
    }

    public void addChild(Renderer child) {
        children.add(child);
    }

    public void removeChild(Renderer child) {
        children.remove(child);
    }

    public List<Renderer> getChildren() {
        return children;
    }

    public void clearChildren() {
        children.clear();
    }

    /**
     * Renders children in sequence.
     *
     * @param renderContext
     *            render context
     */
    protected void renderChildren(RenderContext renderContext) {
        for (Renderer child : children) {
            child.render(renderContext);
        }
    }
}
